src/qmapshack/poi/CPoiPOI.cpp

```diff
@@ -193,19 +194,46 @@ bool CPoiPOI::getToolTip(const QPoint &px, QString &str) const
 
 void CPoiPOI::addTreeWidgetItems(QTreeWidget* widget)
 {
-    QMap<QString, CPoiCategory*> parentMap;
+    // Open database here so it belongs to the right thread
+    if(!QSqlDatabase::contains(filename))
+    {
+        QSqlDatabase db = QSqlDatabase::addDatabase("QSQLITE", filename);
+        db.setDatabaseName(filename);
+        if(!db.open())
+        {
+            qDebug() << "failed to open database" << db.lastError();
+            isActivated = false;
+            return;
+        }
+    }
+
+    QSqlQuery query("SELECT id, name, parent FROM main.poi_categories ORDER BY id DESC", QSqlDatabase::database(filename));
 
-    for(const QString& category : tagMap.keys())
+    QMap<uint, CPoiCategory*> categoryMap;
+
+    while (query.next())
     {
-        QStringList parentChildNameList = category.split("_");
-        QString parentName = parentChildNameList[0];
-        QString childName = parentChildNameList.mid(1).join("_");
-        if(!parentMap.contains(parentName))
+        bool hasParent = false;
+        uint parentID = query.value(eSqlColumnCategoryParent).toUInt(&hasParent);
```

Comment:
Do not use `uint` use the Qt internal `qunit...` types. Database IDs are 64bit types. Please fix all other occurrences, too .